### PR TITLE
refactor(focus): implement two colored outline

### PR DESCRIPTION
**Pull request by mfranzke**
## Proposed changes

Resolves https://github.com/db-ui/mono/issues/1620

## Types of changes

-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [x] Refactoring (fix on existing components or architectural decisions)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation Update (if none of the other choices apply)

## ToDos

- [x] 	Replace color values `#309fd1`, `#fff` and `#242629` by their related design tokens
- [x] Go through each and every element to identify possible collisions

### CODE REVIEW

File: packages/foundations/scss/_init.scss [hunk 1]
```diff
@@ -73,6 +81,8 @@ iframe {
 			--db-focus-outline-offset,
 			var(--db-focus-outline-size)
 		);
+		box-shadow: 0 0 0
```
Comment (nmerget) [inline packages/foundations/scss/_init.scss:84]: We need to add box-shadow to default transition

File: packages/foundations/scss/_init.scss [hunk 2]
```diff
@@ -3,9 +3,9 @@
 
 :root {
 	// TODO: probably move these to the tokens as well
-	--db-focus-outline-offset: 1px;
-	--db-focus-outline-color: #{variables.$db-colors-informational};
-	--db-focus-outline-size: max(2px, 0.08em);
+	--db-focus-outline-offset: 2px;
+	--db-focus-outline-color: #fff;
```
Comment (nmerget) [inline packages/foundations/scss/_init.scss:7]: I don't like that we have a fixed color here. We should ask UX which token we should provide, to make it overwriteable with the theme-builder.

File: packages/foundations/scss/_init.scss (hunk 1 above)
Comment (mfranzke) [reply packages/foundations/scss/_init.scss:84]: Let's align, as I not fully understand what you mean. But in general this declaration is at least still missing a styling API for either further declarations of box-shadow, and/or modifications from the outside besides our necessary declaration for the "outline" defined as a box-shadow, meaning to have e.g. another css variable in this box-shadow-chain.

File: packages/foundations/scss/_init.scss (hunk 2 above)
Comment (mfranzke) [reply packages/foundations/scss/_init.scss:7]: couldn't agree more.

File: packages/foundations/scss/_init.scss (hunk 1 above)
Comment (nmerget) [reply packages/foundations/scss/_init.scss:84]: Sorry I meant the [%default-transition](https://github.com/db-ui/mono/blob/main/packages/foundations/scss/helpers/_component.scss#L34) to animate this new focus state

Comment (mfranzke) [reply packages/foundations/scss/_init.scss]: > Sorry I meant the [%default-transition](https://github.com/db-ui/mono/blob/main/packages/foundations/scss/helpers/_component.scss#L34) to animate this new focus state

@nmerget could you please provide a suggestion or just commit the relevant change?

File: packages/foundations/scss/_init.scss [hunk 3]
```diff
@@ -87,15 +91,43 @@ iframe {
...
+			outline
+				var(
+					--db-transition-duration-ultra-fast,
+					#{variables.$db-transition-duration-ultra-fast}
+				),
+			box-shadow
+				var(
+					--db-transition-duration-ultra-fast,
+					#{variables.$db-transition-duration-ultra-fast}
+				);
```
Comment (mfranzke) [inline packages/foundations/scss/_init.scss:125]: The `outline` and `box-shadow` stylings on `focus-visible` include some fallback values for their CSS Custom Properties, which are currently including these very same CSS Custom Properties as well, which doesn't make any sense.

Comment (mfranzke) [reply packages/foundations/scss/_init.scss:125]: https://github.com/db-ui/mono/issues/1784

File: packages/foundations/scss/_variables.scss [hunk 4]
```diff
@@ -57,3 +59,6 @@ $db-transition-functional-straight: #{$db-transition-duration-medium} #{$db-tran
 
 $db-transition-show-straight: var(--db-transition-show-straight);
 $db-transition-hide-straight: var(--db-transition-hide-straight);
+
+/* Border radius */
+$db-border-radius: helpers.to-rem(4);
```
[the remark this answers is not in the document]
Comment (mfranzke) [reply packages/foundations/scss/_variables.scss:64]: > Maybe the best way would be to move the border radius to the theme builder as well and consume it as a css-property

@nmerget I think that you would need to take care of this aspect, wouldn't you?

[the remark this answers is not in the document]
Comment (nmerget) [reply packages/foundations/scss/_variables.scss:64]: https://github.com/db-ui/mono/pull/1790

Yes :)

**Review by nmerget — APPROVED** [review]

**Pull request merged**